blog/utils/records.ts

```diff
@@ -13,3 +24,171 @@ export async function getRecordsByPath<T>(
+  try {
+    const currentRatings = await records.select({
+      id: rating.id,
+      itemReviewed: rating.itemReviewed,
+      author: rating.author,
+      ratingValue: rating.ratingValue,
+      additionalType: rating.additionalType,
+    })
+      .from(rating).where(eq(rating.itemReviewed, slug)) as
+        | Rating[]
+        | undefined;
+
+    return currentRatings?.length === 0 || !currentRatings
+      ? []
+      : currentRatings.map((rating) => ({
+        ...rating,
+        bestRating: 5,
+        worstRating: 1,
+      }));
+  } catch (e) {
+    logger.error(e);
+    return [];
+  }
+}
+
+export async function getRatings(
+  { ctx, post, ignoreRatings, onlyAggregate }: {
+    ctx: AppContext;
+    post: BlogPost;
+    ignoreRatings?: Ignore;
+    onlyAggregate?: boolean;
+  },
+): Promise<BlogPost> {
+  const contentRating = await getRatingsBySlug({
+    ctx,
+    slug: post.slug,
+  });
+
+  const { ratingCount, ratingTotal } = contentRating.length === 0
+    ? { ratingCount: 0, ratingTotal: 0 }
+    : contentRating.reduce(
+      (acc, { ratingValue, additionalType }) =>
+        ignoreRatings?.active && additionalType &&
+          ignoreRatings.markedAs?.includes(additionalType)
+          ? acc
+          : {
+            ratingCount: acc.ratingCount + 1,
+            ratingTotal: acc.ratingTotal + (ratingValue ?? 0),
+          },
+      { ratingCount: 0, ratingTotal: 0 },
+    );
+
+  const ratingValue = ratingTotal / ratingCount;
+
+  return {
+    ...post,
+    contentRating: onlyAggregate ? undefined : contentRating,
+    aggregateRating: {
+      ...post.aggregateRating,
+      "@type": "AggregateRating",
+      ratingCount,
+      ratingValue: isNaN(ratingValue) ? 0 : ratingValue,
+      bestRating: 5,
+      worstRating: 1,
+    },
+  };
+}
+
+export const getReviewById = async (
+  { ctx, id }: { ctx: AppContext; id?: string },
+): Promise<Review | null> => {
+  if (!id) {
+    return null;
+  }
+  const records = await ctx.invoke.records.loaders.drizzle();
+  try {
+    const targetReview = await records.select({
+      itemReviewed: review.itemReviewed,
+      author: review.author,
+      datePublished: review.datePublished,
+      dateModified: review.dateModified,
+      reviewBody: review.reviewBody,
+      reviewHeadline: review.reviewHeadline,
+      isAnonymous: review.isAnonymous,
+      additionalType: review.additionalType,
+      id: review.id,
+    })
+      .from(review).where(eq(review.id, id)).get() as Review | undefined;
+    return targetReview ?? null;
+  } catch (e) {
+    logger.error(e);
+    return null;
+  }
+};
+
+export async function getReviewsBySlug(
+  { ctx, slug, ignoreReviews, orderBy = "date_desc" }: {
+    ctx: AppContext;
+    slug: string;
+    ignoreReviews?: Ignore;
+    orderBy?: "date_asc" | "date_desc";
+  },
+): Promise<Review[]> {
+  const records = await ctx.invoke.records.loaders.drizzle();
+
+  const whereClause = ignoreReviews?.active && ignoreReviews?.markedAs &&
```

Comment:
Use the name `utils` for things like: `checkIfDateIsBefore`,  `encodeToken`.. and not your core business logic.

We don't have something well setup for that, but you'll come up with a good name